packages/material-ui-lab/src/TreeNode/TreeNode.test.js

```diff
@@ -0,0 +1,555 @@
+  });
+
+  it('should call onFocus when focused', () => {
+    const handleFocus = spy();
+
+    const { getByText } = render(
+      <TreeView>
+        <TreeNode nodeId="test" label="test" onFocus={handleFocus} />
+      </TreeView>,
+    );
+
+    fireEvent.focus(getByText('test'));
+
+    expect(handleFocus.callCount).to.equal(2);
+  });
+
+  it('should call onKeyDown when clicked', () => {
+    const handleKeyDown = spy();
+
+    const { getByText } = render(
+      <TreeView>
+        <TreeNode nodeId="test" label="test" onKeyDown={handleKeyDown} />
+      </TreeView>,
+    );
+
+    fireEvent.keyDown(getByText('test'), { key: 'Enter' });
+
+    expect(handleKeyDown.callCount).to.equal(1);
+  });
+
+  describe('Accessibility', () => {
+    it('should have the role `treeitem`', () => {
+      const { container } = render(
+        <TreeView>
+          <TreeNode nodeId="test" label="test" />
+        </TreeView>,
+      );
+
+      expect(container.querySelector('[role="treeitem"]')).to.be.ok;
+    });
+
+    it('should add the role `group` to a `ul` component if it has children', () => {
+      const { container } = render(
+        <TreeView>
+          <TreeNode nodeId="test" label="test">
+            <TreeNode nodeId="test2" label="test" />
+          </TreeNode>
+        </TreeView>,
+      );
+
+      expect(container.querySelector('ul[role="group"]')).to.be.ok;
+    });
+
+    it('should have the attribute `aria-expanded=false` if collapsed', () => {
+      const { container } = render(
+        <TreeView>
+          <TreeNode nodeId="test" label="test">
+            <TreeNode nodeId="test2" label="test" />
+          </TreeNode>
+        </TreeView>,
+      );
+
+      expect(container.querySelector('[aria-expanded=false]')).to.be.ok;
+    });
+
+    it('should have the attribute `aria-expanded=true` if expanded', () => {
+      const { container } = render(
+        <TreeView expanded={['test']}>
+          <TreeNode nodeId="test" label="test">
+            <TreeNode nodeId="test2" label="test" />
+          </TreeNode>
+        </TreeView>,
+      );
+
+      expect(container.querySelector('[aria-expanded=true]')).to.be.ok;
+    });
+
+    it('should not have the attribute `aria-expanded` if no children are present', () => {
+      const { container } = render(
+        <TreeView>
+          <TreeNode nodeId="test" label="test" />
+        </TreeView>,
+      );
+
+      expect(container.querySelector('[aria-expanded]')).to.not.be.ok;
+    });
+
+    describe('when a tree receives focus', () => {
+      it('should focus the first node if none of the nodes are selected before the tree receives focus', () => {
+        const { getByTestId, getByText } = render(
+          <TreeView>
+            <TreeNode nodeId="1" label="one" data-testid="one" />
+            <TreeNode nodeId="2" label="two" />
+            <TreeNode nodeId="3" label="three" />
+          </TreeView>,
+        );
+
+        fireEvent.focus(getByText('one'));
```

Comment:
Firing a focus event doesn't move focus. The assertion at the end is also true without these actions. Did you mean to `getByText('one').focus()`?